commit 6fcf9affd1cb17bca2c4c4e9129c61d864dda6c9
Author: Gustavo A. R. Silva <gustavoars@kernel.org>
Date:   Wed Jul 22 13:50:24 2020 -0500

    bna: bfi.h: Avoid the use of one-element array
    
    One-element arrays are being deprecated[1]. Replace the one-element
    array with a simple value type 'u8 rsvd'[2], once it seems this is
    just a placeholder for alignment.
    
    [1] https://github.com/KSPP/linux/issues/79
    [2] https://github.com/KSPP/linux/issues/86
    
    Tested-by: kernel test robot <lkp@intel.com>
    Link: https://github.com/GustavoARSilva/linux-hardening/blob/master/cii/0-day/bfi-20200718.md
    Signed-off-by: Gustavo A. R. Silva <gustavoars@kernel.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/ethernet/brocade/bna/bfi.h b/drivers/net/ethernet/brocade/bna/bfi.h
index 09c912e984fe..f780d42c946d 100644
--- a/drivers/net/ethernet/brocade/bna/bfi.h
+++ b/drivers/net/ethernet/brocade/bna/bfi.h
@@ -389,7 +389,7 @@ struct bfi_msgq_mhdr {
 	u16	msg_token;
 	u16	num_entries;
 	u8	enet_id;
-	u8	rsvd[1];
+	u8	rsvd;
 } __packed;
 
 #define bfi_msgq_mhdr_set(_mh, _mc, _mid, _tok, _enet_id) do {	\